src/qnetvo/utilities.py

```diff
@@ -139,3 +139,33 @@ def mixed_base_num(n, base_digits):
         n_tmp = n_tmp % place
 
     return mixed_base_digits
+
+
+def ragged_reshape(input_list, list_dims):
```

Comment:
We have a similar function in PennyLane (https://docs.pennylane.ai/en/latest/code/api/pennylane.utils.unflatten.html), but I think this function may be on it's way to deprecation - so probably fine to add this one.